stellar/deploy-contract.js

```diff
@@ -139,30 +143,71 @@ async function uploadWasm(filePath, wallet, chain) {
 
 async function upgrade(options, _, chain, contractName) {
     const { wasmPath, yes } = options;
-    const contractAddress = chain.contracts[contractName].address;
+    const uncheckedContractAddress = chain.contracts[contractName].address;
     const wallet = await getWallet(chain, options);
 
     if (prompt(`Proceed with upgrade on ${chain.name}?`, yes)) {
         return;
     }
 
     validateParameters({
-        isNonEmptyString: { contractAddress },
+        isNonEmptyString: { uncheckedContractAddress },
     });
 
+    const contractAddress = Address.fromString(uncheckedContractAddress);
+
     const newWasmHash = await uploadWasm(wasmPath, wallet, chain);
     printInfo('New Wasm hash', serializeValue(newWasmHash));
 
     const operation = Operation.invokeContractFunction({
-        contract: contractAddress,
+        contract: chain.contracts.upgrader.address,
         function: 'upgrade',
-        args: [nativeToScVal(newWasmHash)],
+        args: [
+            nativeToScVal(contractAddress),
+            nativeToScVal(options.newVersion),
+            nativeToScVal(newWasmHash),
+            nativeToScVal([options.migrationData]),
+        ],
+        auth: await createUpgradeAuths(contractAddress, newWasmHash, options.migrationData, chain, wallet),
     });
+
     await broadcast(operation, wallet, chain, 'Upgraded contract', options);
     chain.contracts[contractName].wasmHash = serializeValue(newWasmHash);
     printInfo('Contract upgraded successfully!', contractAddress);
 }
 
+async function createUpgradeAuths(contractAddress, newWasmHash, migrationData, chain, wallet) {
+    // 20 seems a reasonable number of ledgers to allow for the upgrade to take effect
```

Comment:
Create a global constant for it. Could also create an option with a default of 20 though I don't see us configuring it